### ISSUE TITLE
Unmarshal invalid unicode characters  in hexadecimal strings

### ISSUE THREAD

**Issue by liuq19** [bytedance/sonic#92]
Sonic does not mark the invalid unicode characters as \ufffd.

```
func TestUnmarshalInvalideUnicodeInHexString(t *testing.T) {
	var v map[string]interface{}
	err := Unmarshal([]byte("{\"x\":\"\xf4\"}"), &v)
    if err != nil {
        println(err.(decoder.SyntaxError).Description())
        require.NoError(t, err)
    }
    require.Equal(t, map[string]interface{}{"x": "\ufffd"}, v)
}
```

Related Golang Specifiction: https://golang.org/ref/spec#Rune_literals

**Comment by liuq19** [comment]
The problem will be solved at #96

**Comment by liuq19** [comment]
You can validate and then unmarshal. #189
